analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Nullable.cs

```diff
@@ -0,0 +1,62 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using SonarAnalyzer.SymbolicExecution.Constraints;
+using SonarAnalyzer.UnitTest.TestFramework.SymbolicExecution;
+
+namespace SonarAnalyzer.UnitTest.SymbolicExecution.Roslyn;
+
+public partial class RoslynSymbolicExecutionTest
+{
+    [TestMethod]
+    public void Nullable_Assignment_PropagatesConstrainsToValue()
+    {
+        const string code = """
+            bool? value = null;
+            Tag("Null", value);
+            value = true;
+            Tag("True", value);
+            """;
+        var validator = SETestContext.CreateCS(code).Validator;
+        validator.ValidateTag("Null", x => x.HasConstraint(ObjectConstraint.Null).Should().BeTrue());
+        validator.ValidateTag("True", x => x.HasConstraint(BoolConstraint.True).Should().BeTrue());
+    }
+
+    [TestMethod]
+    public void Nullable_Value_ReadsConstraintsFromInstance()
+    {
+        const string code = """
+            var value = arg.Value;
+            Tag("Unknown", value);
```

Comment:
I suppose this will change to `Tag("NoNull", value);` in the future.

Reply:
Yes, that will most likely happen on your PR.

Reply:
Bingo: https://dev.azure.com/sonarsource/DotNetTeam%20Project/_build/results?buildId=63594&view=ms.vss-test-web.build-test-results-tab&runId=1244981&resultId=100793&paneView=debug

Reply:
💥